### DISCUSSION TITLE
how to create http route for all incoming messages with all domain

### DISCUSSION THREAD

**Discussion by Muhammedalihudaybergenow** [postalserver/postal#2485] [Development discussions]
I want to create route that listens all incoming messages from postal incoming messages

**Comment by willpower232** [comment]
do you mean you want to set up one route that applies to all Organisations and Servers so when anything emails your Postal server you get it?

**Reply by Muhammedalihudaybergenow** [reply]
Mail server has an organization and a domain. To send messages works fine that means the dns configuration has been defined. I have sent messages via HTTP API. I am trying to do incoming messages too. But I have to create http routes for each user. Should I need to create a separate route for each user? Is there any way to listen to all incoming messages in a single route?

![Снимок экрана 2023-06-05 132328](https://github.com/postalserver/postal/assets/94282427/156f495c-d0b1-4f61-999d-59571b569b96)

I want to use such *@example.com and to listen all incoming messages to all route name redirect to my backend server and process it

**Reply by willpower232** [reply]
the screenshot will route all email addressed to anything@yonekey.site to that http endpoint, it sounds like you have already achieved what you want?

**Reply by Muhammedalihudaybergenow** [reply]
![Снимок экрана 2023-06-06 115440](https://github.com/postalserver/postal/assets/94282427/1a34226f-2dee-45ad-a60d-a51edc374f43)
but i have this kind of error when sending mail from external sources

**Reply by willpower232** [reply]
which means that there are no DNS records for mx.postaldemo.yonekey.site

**Reply by Muhammedalihudaybergenow** [reply]
but if i send message to postal and when registered routes as above i have such kind of errors:
<img width="1183" alt="Screenshot 2023-06-11 at 5 39 38 PM" src="https://github.com/postalserver/postal/assets/94282427/53856e6d-0af6-41ce-b128-48bcd654b4a8">
<img width="1188" alt="Screenshot 2023-06-11 at 5 39 13 PM" src="https://github.com/postalserver/postal/assets/94282427/4e535db6-9c72-4b17-a206-378ec2ed86c4">

**Comment by Muhammedalihudaybergenow** [comment]
My dns configuration is showing than i haven't got any errors
<img width="1133" alt="Screenshot 2023-06-11 at 5 45 09 PM" src="https://github.com/postalserver/postal/assets/94282427/97ad91b9-a5b2-479b-a6cd-f10cd35955d4">
<img width="621" alt="Screenshot 2023-06-11 at 5 44 40 PM" src="https://github.com/postalserver/postal/assets/94282427/6eebdfe8-7e92-4181-99a0-9a7dfa44b9ad">

**Reply by sinnbeck** [reply]
You set the mx to point to mx.postaldemo.yonekey.site 

But that does not have any a record?
https://dnskit.dk/mx.postaldemo.yonekey.site

That should probably point to your postal server ip

**Reply by willpower232** [reply]
mx.postaldemo.yonekey.site should be an A record pointing at your Postal server

then other domains can have an MX record with a value of mx.postaldemo.yonekey.site.

Postal has correctly evaluated the textual portions of the DNS records but does not evaluate the IP addresses behind them so it can't see that there is an issue.
